internal/libyaml/writer.go

```diff
@@ -22,27 +22,22 @@
 
 package libyaml
 
-// Set the writer error and return false.
-func (emitter *Emitter) setWriterError(problem string) bool {
-	emitter.ErrorType = WRITER_ERROR
-	emitter.Problem = problem
-	return false
-}
+import "fmt"
 
 // Flush the output buffer.
-func (emitter *Emitter) flush() bool {
+func (emitter *Emitter) flush() error {
 	if emitter.write_handler == nil {
 		panic("write handler not set")
```

Comment:
Just a comment. I think we need to get rid of the panic where it is possible and return error instead. But this is going to be a huge work, and should be done after all active PRs are merged.